core/vcpu.c

```diff
@@ -1189,6 +1188,10 @@ void vcpu_save_host_state(struct vcpu_t *vcpu)
         vmwrite(vcpu, HOST_EFER, hstate->_efer);
     }
 
+    if (vmx(vcpu, exit_ctls) & EXIT_CONTROL_LOAD_PAT) {
+        vmwrite(vcpu, HOST_PAT, ia32_rdmsr(IA32_CR_PAT));
```

Comment:
Can we add ia32_pat to hstate so that more debug friendly?

Reply:
What means more debug friendly?

Reply:
From the context I'd say (@coxuintel correct me if I'm wrong) something analogous to the few *IA32_EFER*-related lines above, i.e. first storing the `IA32_CR_PAT` value in *hstate*:
```
    hstate->_pat = ia32_rdmsr(IA32_CR_PAT);
```
and then vmwriting directly from that field:
```
    if (vmx(vcpu, exit_ctls) & EXIT_CONTROL_LOAD_PAT) {
        vmwrite(vcpu, HOST_PAT, hstate->_pat);
    }
```

Reply:
Exactly. Then functions like vcpu_get_host_state() could output more information then.